From c9bf6c0b826dc5d4bbaa758063e66bfb2978675e Mon Sep 17 00:00:00 2001
Message-Id: <patch-3.6-c9bf6c0b826-20210906T072851Z-avarab@gmail.com>
In-Reply-To: <96814e5e-54be-1eca-0d75-68be53b1be3d@archlinux.org>
References: <96814e5e-54be-1eca-0d75-68be53b1be3d@archlinux.org>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Thu, 15 Jul 2021 16:44:31 +0200
Subject: [PATCH 3/6] shortlog: use designated initializer for "struct
 shortlog"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change code added in 64093fc06a (blame,shortlog: don't make local
option variables static, 2016-06-13) to use a designated initializer
via a typical *_INIT macro pattern.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 builtin/shortlog.c | 2 +-
 shortlog.h         | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/builtin/shortlog.c b/builtin/shortlog.c
index 3e7ab1ca821..fa1f76cc51e 100644
--- a/builtin/shortlog.c
+++ b/builtin/shortlog.c
@@ -342,7 +342,7 @@ void shortlog_init(struct shortlog *log)
 
 int cmd_shortlog(int argc, const char **argv, const char *prefix)
 {
-	struct shortlog log = { STRING_LIST_INIT_NODUP };
+	struct shortlog log = SHORTLOG_INIT;
 	struct rev_info rev;
 	int nongit = !startup_info->have_repository;
 
diff --git a/shortlog.h b/shortlog.h
index 3f7e9aabcae..47892d6d604 100644
--- a/shortlog.h
+++ b/shortlog.h
@@ -28,6 +28,10 @@ struct shortlog {
 	FILE *file;
 };
 
+#define SHORTLOG_INIT { \
+	.list = STRING_LIST_INIT_NODUP, \
+}
+
 void shortlog_init(struct shortlog *log);
 
 void shortlog_add_commit(struct shortlog *log, struct commit *commit);
-- 
2.33.0.821.gfd4106eadbd

